src/main/scala/fi/oph/koski/validation/KoskiValidator.scala

```diff
@@ -390,6 +390,15 @@ class KoskiValidator(tutkintoRepository: TutkintoRepository, val koodistoPalvelu
       } else {
         HttpStatus.ok
       }
+
+    case ib: IBTutkinnonSuoritus =>
+      def viimeisinArviointiNumeerinen(suoritus: Suoritus) = suoritus.arviointi.flatMap(_.lastOption.map(_.arvosana.koodiarvo)).map(_ forall isDigit)
+
+      HttpStatus.fold(ib.osasuoritusLista
+        .groupBy(_.koulutusmoduuli.identiteetti)
+        .collect { case (koodiarvo , osasuoritukset) if osasuoritukset.flatMap(viimeisinArviointiNumeerinen).count(_ == true) > 1 => koodiarvo }
```

Comment:
Ehkä selkeämpää ilman `flatMap`:iä. Joku metodi joka tyyppiä Suoritus => Boolean on parempi niin sitten voi tehdä tarkistuksen näin:
```
if osasuoritukset.count(hasNumeerinenArviointi) > 1) => koodiarvo
```